src/Mvc/Mvc.Core/test/HttpActionResultTests.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+
+namespace Microsoft.AspNetCore.Mvc;
+
+public class HttpActionResultTests
+{
+    [Fact]
+    public void HttpActionResult_InitializesWithResultsStaticMethods()
+    {
+        // Arrange & Act
+        var httpResult = Mock.Of<IResult>();
+        var result = new HttpActionResult(httpResult);
+
+        // Assert
+        Assert.Equal(httpResult, result.Result);
+    }
+
+    [Fact]
+    public async Task HttpActionResult_SetsStatusCode()
```

Comment:
Is the assert missing, or did I not notice it?

Reply:
I changed the name, `HttpActionResult_InvokesInternalHttpResult`,  to make it clearer, but the "assertion" happens on the call `Verify()` since the only thing that the `ActionResult` does is call the internal IResult instance.

Here some examples of similar tests:
https://github.com/dotnet/aspnetcore/blob/ead83983d38d796f136eb6c1a322afd37a1d1da1/src/Http/Http.Results/test/ForbidResultTest.cs#L16

Reply:
I changed the method name, in this last commit, because I forgot to do it before, since the original code I was using a concrete `IResult` class instead of the mock.